sqlitecluster/SQLiteNode.cpp

```diff
@@ -458,6 +458,22 @@ list<STable> SQLiteNode::getPeerInfo() const {
     return peerData;
 }
 
+string SQLiteNode::getEligibleFollowerForForwarding() const {
+    vector<string> validPeers;
+    const string leaderVersion = getLeaderVersion();
+    for (SQLitePeer* peer : _peerList) {
+        // We want only nodes that are using the same version as leader and are currently followers
+        if (peer->version.load() != leaderVersion || peer->state.load() != SQLiteNodeState::FOLLOWING) {
+            continue;
+        }
+        validPeers.push_back(peer->name);
+    }
+    if (validPeers.empty()) {
+        return "";
+    }
+    return validPeers[rand() % validPeers.size()];
```

Comment:
Does this always return the same thing?

Reply:
no, it will do a random to get it between the current valid active peers. During the tests, I had a few set of different peers return.

Reply:
You sure? because I just made this change to the bedrock tests:
```
int main(int argc, char* argv[]) {
    cout << rand() << endl;
    cout << rand() << endl;
    cout << rand() << endl;
    cout << rand() << endl;
    cout << rand() << endl;
    return 0;
````

And running it multiple times does this:
```
vagrant@expensidev2004:/vagrant/Bedrock/test$ ./test
1804289383
846930886
1681692777
1714636915
1957747793
vagrant@expensidev2004:/vagrant/Bedrock/test$ ./test
1804289383
846930886
1681692777
1714636915
1957747793
vagrant@expensidev2004:/vagrant/Bedrock/test$ ./test
1804289383
846930886
1681692777
1714636915
1957747793
vagrant@expensidev2004:/vagrant/Bedrock/test$ ./test
1804289383
846930886
1681692777
1714636915
1957747793
```

Reply:
I don't know if this really matters much because the randomness here isn't that important, but the C `rand()` function is not very random.

Reply:
But every time we call random on the same instance after it was built, it will return separate values. It will be always the same order, but we should see different servers everytime something is escalated.

I do agree that that's not the most important thing as well.